src/math/mat.zig

```diff
@@ -331,129 +75,230 @@ pub fn Mat4x4(
+        pub inline fn scale(s: VecMin1) Self {
+            var result: Self = .ident;
+            inline for (0..@TypeOf(s).n) |i| {
+                result.v[i].v[i] *= s.v[i];
+            }
+            return result;
         }
 
-        /// Constructs a 3D matrix which scales each dimension by the given scalar.
-        pub inline fn scaleScalar(s: Vec.T) Matrix {
-            return scale(math.Vec3.splat(s));
+        /// Constructs a matrix which scales each dimension by the given scalar.
+        pub inline fn scaleScalar(t: T) Self {
+            return scale(.splat(t));
         }
 
-        /// Constructs a 3D matrix which translates coordinates by the given vector.
-        pub inline fn translate(t: math.Vec3) Matrix {
-            return init(
-                &RowVec.init(1, 0, 0, t.x()),
-                &RowVec.init(0, 1, 0, t.y()),
-                &RowVec.init(0, 0, 1, t.z()),
-                &RowVec.init(0, 0, 0, 1),
-            );
+        /// Constructs a matrix which translates coordinates by the given vector.
+        pub inline fn translate(t: VecMin1) Self {
+            var result: Self = .ident;
+            inline for (0..@TypeOf(t).n) |i| {
+                result.v[cols - 1].v[i] += t.v[i];
+            }
+            return result;
         }
 
-        /// Constructs a 3D matrix which translates coordinates by the given scalar.
-        pub inline fn translateScalar(t: Vec.T) Matrix {
-            return translate(math.Vec3.splat(t));
+        /// Constructs a 1D matrix which translates coordinates by the given scalar.
+        pub inline fn translateScalar(t: T) Self {
+            return translate(.splat(t));
         }
 
-        /// Returns the translation component of the matrix.
-        pub inline fn translation(t: *const Matrix) math.Vec3 {
-            return math.Vec3.init(t.v[3].x(), t.v[3].y(), t.v[3].z());
+        pub inline fn translation(t: *const Self) VecMin1 {
+            var result: VecMin1 = undefined;
+            inline for (0..VecMin1.n) |i| {
+                result.v[i] = t.v[cols - 1].v[i];
+            }
+            return result;
         }
 
         /// Constructs a 3D matrix which rotates around the X axis by `angle_radians`.
-        pub inline fn rotateX(angle_radians: f32) Matrix {
+        pub inline fn rotateX(angle_radians: f32) Mat4x4(T) {
             const c = math.cos(angle_radians);
             const s = math.sin(angle_radians);
-            return Matrix.init(
-                &RowVec.init(1, 0, 0, 0),
-                &RowVec.init(0, c, -s, 0),
-                &RowVec.init(0, s, c, 0),
-                &RowVec.init(0, 0, 0, 1),
-            );
+            return .init(.{
+                &.init(.{ 1, 0, 0, 0 }),
+                &.init(.{ 0, c, -s, 0 }),
+                &.init(.{ 0, s, c, 0 }),
+                &.init(.{ 0, 0, 0, 1 }),
+            });
         }
 
         /// Constructs a 3D matrix which rotates around the X axis by `angle_radians`.
```

Comment:
```suggestion
        /// Constructs a 3D matrix which rotates around the Y axis by `angle_radians`.
```
Looks like there's been a typo here.

Reply:
Fixed :+1: